packages/contracts/test/rbac.ts

```diff
@@ -0,0 +1,460 @@
+import { Provider, ProviderRegistry } from "@blockstack/clarity";
+import { expect } from "chai";
+import { RBACClient } from "../src/rbac-client"
+import { Clarity } from "../src/util"
+
+
+
+describe("rbac contract test suite", () => {
+  let provider: Provider;
+  let client: RBACClient;
+  const DEPLOYER = 'SP3GWX3NE58KXHESRYE4DYQ1S31PQJTCRXB3PE9SB';
+  const USER = 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6';
+
+
+  describe("Contract syntax", async () => {
+    it("should be valid", async () => {
+      provider = await ProviderRegistry.createProvider();
+      client = new RBACClient(DEPLOYER, provider);
+
+      await client.checkContract();
+      await client.deployContract();
+    });
+
+    after(async () => {
+      await provider.close();
+    });
+  });
+
+  describe("Contract logic [Roles]", async () => {
+    beforeEach(async () => {
+      provider = await ProviderRegistry.createProvider();
+      client = new RBACClient(DEPLOYER, provider);
+
+      await client.deployContract();
+    });
+
+   
+    afterEach(async () => {
+      await provider.close();
+    });
+
+
+    it("returns false if user doesn't have any roles", async () => {
+      const invalidRole = 129;
+      const receipt = await client.hasRole(USER, invalidRole);
+
+      expect(Clarity.unwrapBool(receipt),`${USER} should not have role: ${invalidRole}.`).to.be.false;
+    });
+
+   
+    it("should fail granting role higher than 127", async () => {
+      const expectedError = "Role is out of range 0-127";
+
+      for (let role = 128; role <=150; role++) {
+        let receipt = await client.grantRole(USER, role, DEPLOYER);
+
+        expect(receipt.success,`The success status of granting role ${role} should be false.`).to.be.false;
+        
+        expect(Clarity.unwrapError(receipt), 
+          `Granting role ${role} should fail with "${expectedError}" error.`).to.be.eq(expectedError);
+      }
+    });
+
+   
+    it("should succeed granting role in range 0-127", async () => {
+      for (let role = 0; role <= 127; role++) {
+        let receipt = await client.grantRole(USER, role, DEPLOYER);
+      
+        expect(receipt.success, `The success status of granting role ${role} should be true.`).to.be.true;
+      }
+    });
+
+    
+    it("should remember granted role", async () => {
+      const roles = [2, 8 , 18, 55, 37, 115];
+
+      for (const role of roles) {
+        await client.grantRole(USER, role, DEPLOYER);
+        
+        let receipt = await client.hasRole(USER, role);
+
+        expect(Clarity.unwrapBool(receipt), 
+          `${USER} should have role ${role} after it has been granted.`).to.be.true;
+      }
+    });
+
+
+    it("tests if user has a specific role", async() => {
```

Comment:
"should return false if user has not specific role"

Reply:
This test is redundant as it tests exactly the same thing as the one before `should remember granted role`.
The only difference is that at the beginning of this one we check if user have no roles assigned which is pointless.

I'll remove it.